SECURITY.md

```diff
@@ -0,0 +1,5 @@
+## Reporting a Vulnerability
```

Comment:
Won't adding this file override the awslabs default? The default is already showing up for this repo: https://github.com/awslabs/smithy-rs/security/policy